services/business/checkers/route_rules/version_presence_checker_test.go

```diff
@@ -0,0 +1,124 @@
+package route_rules
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+func TestCheckerWithPodsMatching(t *testing.T) {
```

Comment:
I suggest to add some basic test with inputs with nil, or empty objects to ensure things don't crash.